tests/posix/common/src/stropts.c

```diff
@@ -18,6 +18,19 @@ ZTEST(stropts, test_putmsg)
 	zassert_equal(errno, ENOSYS, "Expected errno ENOSYS, got %d", errno);
 }
 
+ZTEST(stropts, test_putpmsg)
+{
+	const struct strbuf *ctrl = NULL;
+	const struct strbuf *data = NULL;
+	int fd = -1;
+	int band = 0;
+	int flags = 0;
+	int ret = putpmsg(fd, ctrl, data, band, flags);
+
+	zassert_equal(ret, -1, "Expected return value -1, got %d", ret);
+	zassert_equal(errno, ENOSYS, "Expected errno ENOSYS, got %d", errno);
+}
```

Comment:
Normally, it's a great idea to permute through the dynamic range of input, so usually I would say this is a good layout. However, it would be better if the function were implemented and testable.

For stubs, let's keep it simple.

```suggestion
ZTEST(stropts, test_putpmsg)
{
		zassert_equal(putpmsg(-1, NULL, NULL, 0, 0), -1, "Expected return value -1, got %d", ret);
	zassert_equal(errno, ENOSYS, "Expected errno ENOSYS, got %d", errno);
}
```